File: src/types/task-lifecycle.ts
export interface TaskRun {
  id: string;
  workspaceId: string;
  kind: string;
  status: string;
  sourceId?: string | null;
  startedAt: string;
  endedAt?: string | null;
}

export interface TaskEvent {
  id: string;
  taskRunId: string;
  workspaceId: string;
  ts: string;
  eventType: string;
  payload: Record<string, unknown>;
}

export interface WorkspaceTaskSnapshot {
  workspaceId: string;
  runs: TaskRun[];
  events: TaskEvent[];
}

export interface WorkspaceSchedulerJob {
  id: string;
  workspaceId: string;
  kind: string;
  intervalSeconds: number;
  nextRunAt: number;
  enabled: boolean;
  jitterPct: number;
}

File: src-tauri/src/models/task_lifecycle.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TaskRun {
    pub id: String,
    pub workspace_id: String,
    pub kind: String,
    pub status: String,
    pub source_id: Option<String>,
    pub started_at: String,
    pub ended_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TaskEvent {
    pub id: String,
    pub task_run_id: String,
    pub workspace_id: String,
    pub ts: String,
    pub event_type: String,
    pub payload: serde_json::Value,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceTaskSnapshot {
    pub workspace_id: String,
    pub runs: Vec<TaskRun>,
    pub events: Vec<TaskEvent>,
}
